lib/setup.js

```diff
@@ -2135,20 +2164,31 @@ function restartController(callback) {
     }
 }
 
-function installNpm(adapter, callback) {
+function installNpm(adapter, executeRebuild, callback) {
+    if (typeof executeRebuild === 'function') {
+        callback = executeRebuild;
+        executeRebuild = false;
+    }
+
     let path = __dirname;
     if (typeof adapter === 'function') {
         callback = adapter;
         adapter = undefined;
     }
 
     if (adapter) {
-        path = tools.getAdapterDir(adapter);
+        if (executeRebuild && adapter === 'self') {
```

Comment:
Not sure about `rebuild self`... is this ever necessary?

Reply:
I think we will see that. In fact unix-dgram needed for syslog support is a binary module ... I need to test what happens with a nodejs update then ... ok ... on my testsystem I also update nodejs and unix.dgram still works.

Leave it for the case we once need it? Or add it when?

Reply:
Guess it doesn't hurt to leave it in